### PR TITLE
Add profile information columns to user table

### LINKED ISSUE #16: Database table for User Profile Information
### Pre-requisites

#6 (Create Database for user login information) for User's ID column

### Is your feature request related to a problem? Please describe.

Need a database to store the user's information that will be displayed on their profile page.

### Describe the solution you'd like

**Table 1: User Identification**
Columns:
- User's ID (GitHub id?): int [primary key, foreign key from user login table, non-null]
- Name: varchar
- Account_Type: varchar [non-null]
- Pronouns: varchar
- Location: varchar
- Is Looking for Job status: boolean (or int limited to 0 or 1)
- Reason for joining: varchar
- GitHub Link: varchar
- LinkedIn Link: varchar

**Table 2: Student Information**
Columns:
-	User's ID (GitHub id?): int [primary key, foreign key from user login table, non-null]
-	School_Name: varchar
- 	Major: varchar
- 	Graduation Date (month/yr): date 

**Table 3: User Skills**
Columns:
-	User’s ID (GitHub id?): int [primary key, foreign key from user login table, non-null]
-	Language/Frameworks: varchar

**Table 4: User Certifications**
Columns: 
-	User’s ID (GitHub id?): int [primary key, foreign key from user login table, non-null]
-	Certifications: varchar

Table 3 and Table 4 exist for normalization purposes because it is possible to have multiple values for Languages/Frameworks and Certifications.

### Describe alternatives you've considered

- Instead of creating separate tables for Language/Frameworks/Certifications, maybe just concatenate all of the language/frameworks and certifications entered into their own giant varchars (one for languages/frameworks, one for certifications) and store them as their own columns in Table 1.
	- Downside: Might be harder to search for a specific skill(s) as you’ll have to search through the giant string instead of the skill being by itself. (In other words, it might be harder to search through the string instead of just a “SELECT * WHERE Language/Frameworks = “React”). I don't have enough experience with SQL (if we’re using Relational Databases) to know if this isn’t an issue.

- Additional columns:
	- Table 1:
		- Profile pictures: varchar
			- Comment: Do we have anywhere to store the pictures? Should we just use the same picture as what's currently on their GitHub account?

### Additional context

- I'm not sure how big the columns should be.

### PR DESCRIPTION
**Pull request by ayushagarwal0316**
Adding profile information columns to user table:

### Pre-Requisites

- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevant code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

### Description of Changes

Added columns to user table:
pronouns: text not null,
location: text,
hireable: boolean,
purpose: text not null,
schoolName: text not null,
major: text not null,
graduationDate: date not null

### Related Issues

Closes #16

### CODE REVIEW

File: packages/api/src/migrations/0000-add-the-user-table.ts [hunk 1]
```diff
@@ -3,6 +3,8 @@ import { Migration } from '@mikro-orm/migrations';
 export class Migration20210927234531 extends Migration {
   async up(): Promise<void> {
     this.addSql('create table "user" ("id" bigserial primary key, "createdAt" timestamptz(0) not null default clock_timestamp(), "updatedAt" timestamptz(0) not null default clock_timestamp(), "name" text not null);');
+    //in above query, add fields: 
+    //"pronouns" text not null, "location" text, "hireable" boolean, "purpose" text not null, "schoolName" text not null, "major" text not null, "graduationDate" date(0) not null
```
Comment (ayushagarwal0316) [inline packages/api/src/migrations/0000-add-the-user-table.ts:7]: Which fields should be specified not null?
Is naming convention good?

File: packages/api/src/entities/User.ts [hunk 2]
```diff
@@ -9,10 +9,19 @@ export type UserConstructorValues = ConstructorValues<User>;
...
   @Property({ columnType: 'text' })
   name: string;
+  //pronouns: string;
+  //location: string;
+  //hireable: boolean;
+  //purpose: string;
+  //schoolName:string;
+  //major: string;
+  //graduationDate: Date
+
```
Comment (ayushagarwal0316) [inline packages/api/src/entities/User.ts:19]: Would this change be needed for the database side or would this be on the api PR #27 ?
How does typescript Date work with PostgreSQL date type?

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]
@ayushagarwal0316 I think we didn't do a great job of explaining the DB stuff:
- The entity file for a given table _**IS**_ the table as far as you're concerned:
  - First you make an entity file
  - Next you run the commands in the README to generate a `migration`
  - You need to add a `down` method because MicroORM won't make it for you
- On changes to an entity, you modify the entity file, not the migration
- Also remember you should NEVER modify a migration after they've been created (aside from adding the `down`)
  - Migrations are only ever executed once for a DB - they get triggered and then the content they contain is applied to the DB
  - Modifying a migration in the past will execute changes out of order (migration A is run, migration B is run, and then you modify A -> that change won't ever run for your but for a new user, the changes to A may not make sense if they address things changed/created in B)

Please revert all these changes and modify the `User` entity to add the new fields to the class and to the constructor

**Comment by SpencerKaiser** [comment]
@ayushagarwal0316 after reviewing your other PR, please just modify the original entity and then run the command to generate a new migration after undoing the one you already executed for your local DB.

If you have any questions, please post in Slack!

**Pull request closed without merging**